src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -50,13 +50,14 @@ const logStateTransition = (
 ) => {
   if (newState.logs.length > oldState.logs.length) {
     newState.logs.slice(oldState.logs.length).forEach((log) => {
-      const getLogger = (level: keyof Logger) => {
+      const doLog = (level: MigrationLogLevel, message: string) => {
         if (level === 'error') {
-          return logger[level] as Logger['error'];
+          logger.error(message);
+        } else {
+          logger[level](message);
         }
-        return logger[level] as Logger['info'];
       };
-      getLogger(log.level)(logMessagePrefix + log.message);
+      doLog(log.level, logMessagePrefix + log.message);
```

Comment:
AFAIK we only got new state messages when something bad appends, which is why no integration tests caught that. Unit tests were also blind because we're using a mocked logger which is not impacted by this problem (see https://github.com/elastic/kibana/issues/98406#issuecomment-827381625)